rustenv.py

```diff
@@ -59,7 +59,7 @@
 RUSTENV_PROXY = '''\
 #!/bin/sh
 set -eu
-rustinstall=$(dirname $(dirname $(readlink -f "$0")))/rust
+rustinstall="$(cd -L "$(dirname "$(dirname "$0")")" && pwd)/rust"
```

Comment:
I think this is supposed to be `"$(dirname "$(cd -L "$(dirname "$0")" && pwd)")/rust"` -- the goal being to follow as much of whatever symlinks are there as possible

I also don't have commit bit on this repo so I can't merge / update the PR directly

Reply:
Thank you @asottile for this comment.

Just to b sure I understand correctly

I suggested

```shell
$(cd -L "$(dirname "$(dirname "$0")")" && pwd)
```

Which means

> move to the grand-parent directory of `$0`, forcing symbolic links to be followed, and print this cwd

You suggests

```shell
$(dirname "$(cd -L "$(dirname "$0")" && pwd)")
```

Which means 

> move to the directory where `$0` is, forcing symbolic links to be followed, and print the parent of this cwd

Basically, by resolving the symlink earlier, we can then assume the whole path is a real one with a minimum of operations on possibly symbolic parts.  That makes sense to me.  Thank you!